app/javascript/miq-redux/reducer.ts

```diff
@@ -0,0 +1,25 @@
+import {createStore, Store, Reducer, Action} from 'redux';
+
+import { IMiqAction, AppReducer, IMiqReducerHash, AppState} from '../packs/miq-redux';
+
+const reducers: Set<AppReducer> = new Set();
+
+export const rootReducer: AppReducer = (state = {}, action: IMiqAction) => {
+    let newState = state;
+
+    reducers.forEach((reducer) => {
+        newState = reducer(newState, action)
```

Comment:
@vojtechszocs what do you think about this pattern instead? https://medium.com/@SntsDev/the-factory-pattern-in-js-es6-78f0afad17e9